app/controllers/home-office-details.ts

```diff
@@ -88,14 +88,14 @@ function postDateLetterSent(updateAppealService: UpdateAppealService) {
         month,
         year
       };
-
-      await updateAppealService.submitEvent(Events.EDIT_APPEAL, req);
-
       if (diffInDays <= 14) {
         req.session.appeal.application.isAppealLate = false;
       } else {
         req.session.appeal.application.isAppealLate = true;
       }
+
+      await updateAppealService.submitEvent(Events.EDIT_APPEAL, req);
+
```

Comment:
I thought you said we need to update CCD first as otherwise you could update your session, then fail to update CCD which will give you an error page. Then reload and the site will display the update but its not saved in CCD. Or was there an even better reason to do it this way round.